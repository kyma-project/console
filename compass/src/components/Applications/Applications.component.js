import React from 'react';
import PropTypes from 'prop-types';
import { Counter } from 'fundamental-react/Badge';
import LuigiClient from '@kyma-project/luigi-client';

import CreateApplicationModal from './CreateApplicationModal/CreateApplicationModal.container';
import StatusBadge from '../Shared/StatusBadge/StatusBadge';
import { GenericList, handleDelete } from 'react-shared';
import { EMPTY_TEXT_PLACEHOLDER } from '../../shared/constants';
import ScenariosDisplay from './../Shared/ScenariosDisplay/ScenariosDisplay';
import { PageHeader } from 'react-shared';

class Applications extends React.Component {
  static propTypes = {
    applications: PropTypes.object.isRequired,
  };

  headerRenderer = applications => [
    'Name',
    'Provider name',
    'Description',
    'Scenarios',
    'API Definitions',
    'Event Definitions',
    'Status',
  ];

  rowRenderer = application => [
    <span
      className="link"
      onClick={() =>
        LuigiClient.linkManager().navigate(`details/${application.id}`)
      }
    >
      {application.name}
    </span>,
    application.providerName
      ? application.providerName
      : EMPTY_TEXT_PLACEHOLDER,
    application.description ? application.description : EMPTY_TEXT_PLACEHOLDER,
    <ScenariosDisplay scenarios={application.labels.scenarios || []} />,
    <Counter>{application.apiDefinitions.totalCount}</Counter>,
    <Counter>{application.eventDefinitions.totalCount}</Counter>,
    <StatusBadge
      status={
        application.status && application.status.condition
          ? application.status.condition
          : 'UNKNOWN'
      }
    />,
  ];

  actions = [
    {
      name: 'Delete',
      handler: entry => {
        handleDelete(
          'Application',
          entry.id,
          entry.name,
          this.props.deleteApplication,
          this.props.applications.refetch,
        );
      },
    },
  ];

  render() {
    const applicationsQuery = this.props.applications;

    const applications =
      (applicationsQuery &&
        applicationsQuery.applications &&
        applicationsQuery.applications.data) ||
      {};
    const loading = applicationsQuery && applicationsQuery.loading;
    const error = applicationsQuery && applicationsQuery.error;

    if (loading) return 'Loading...';
    if (error) return `Error! ${error.message}`;

    return (
      <>
        <PageHeader title="Applications" />
        <GenericList
          extraHeaderContent={
            <CreateApplicationModal applicationsQuery={applicationsQuery} />
          }
          actions={this.actions}
          entries={applications}
          headerRenderer={this.headerRenderer}
          rowRenderer={this.rowRenderer}
        />
      </>
    );
  }
}

export default Applications;
