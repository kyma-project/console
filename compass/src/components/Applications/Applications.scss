.create-application-dropdown {
  max-width: 250px;

  li {
    list-style-type: none;
  }
}
