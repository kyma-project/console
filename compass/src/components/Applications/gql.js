import gql from 'graphql-tag';

export const GET_APPLICATIONS = gql`
  query {
    applications {
      data {
        id
        providerName
        name
        description
        labels
        status {
          condition
        }
        apiDefinitions {
          totalCount
        }
        eventDefinitions {
          totalCount
        }
      }
    }
  }
`;

export const REGISTER_APPLICATION_MUTATION = gql`
  mutation registerApplication($in: ApplicationRegisterInput!) {
    registerApplication(in: $in) {
      name
      providerName
      description
      labels
      id
    }
  }
`;

export const UNREGISTER_APPLICATION_MUTATION = gql`
  mutation unregisterApplication($id: ID!) {
    unregisterApplication(id: $id) {
      name
      providerName
      description
      labels
      id
    }
  }
`;

export const CHECK_APPLICATION_EXISTS = gql`
  query applications($filter: [LabelFilter!]) {
    applications(filter: $filter) {
      data {
        name
      }
    }
  }
`;
