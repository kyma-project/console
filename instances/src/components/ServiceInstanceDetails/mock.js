export const Mock = {
  __typename: 'clusterDocsTopic',
  name: 'kyma',
  groupName: 'root',
  displayName: 'Kyma',
  description: 'Overall documentation for Kyma',
  assets: [
    {
      __typename: 'clusterAsset',
      name: 'asset1',
      type: 'markdown',
      files: [
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/01-01-in-a-nutshell.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'In a nutshell',
            type: 'Overview',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-01-overview.md',
          metadata: {
            __typename: 'fileMetadata',
            title: 'Overview',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/04-03-cluster-installation.md',
          metadata: {
            __typename: 'fileMetadata',
            // title: 'Install Kyma on a cluster',
            type: 'Installation',
          },
        },
        {
          __typename: 'file',
          url:
            'https://raw.githubusercontent.com/kyma-project/kyma/master/docs/kyma/docs/12-01-try-out-kyma.md',
          metadata: {
            __typename: 'fileMetadata',
            type: 'Examples',
            title: 'Try out Kyma',
          },
        },
      ],
    },
  ],
};
