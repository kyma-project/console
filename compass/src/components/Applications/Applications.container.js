import { graphql, compose } from 'react-apollo';
import { GET_APPLICATIONS, DELETE_APPLICATION_MUTATION } from './gql';

import Applications from './Applications.component';

export default compose(
  graphql(GET_APPLICATIONS, {
    name: 'applications',
    options: {
      fetchPolicy: 'cache-and-network',
    },
  }),
  graphql(DELETE_APPLICATION_MUTATION, {
    props: ({ mutate }) => ({
      deleteApplication: id =>
        mutate({
          variables: {
            id: id,
          },
        }),
    }),
  }),
)(Applications);
