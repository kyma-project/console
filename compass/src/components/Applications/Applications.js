import React from 'react';
import { Token } from '@kyma-project/react-components';
import { Badge, Counter } from 'fundamental-react/lib/Badge';

import GenericList from '../../shared/components/GenericList/GenericList';
import { Query } from 'react-apollo';
import { GET_APPLICATIONS } from './gql';
import LuigiClient from '@kyma-project/luigi-client';
import CreateApplicationModal from './CreateApplicationModal/CreateApplicationModal.container';

class Applications extends React.Component {
  createLabels = labels => {
    const separatedLabels = [];
    for (const key in labels) {
      if (labels.hasOwnProperty(key) && labels[key].length > 0) {
        labels[key].forEach(lab => {
          if (lab === 'undefined') {
            separatedLabels.push(key);
          } else {
            separatedLabels.push(key + '=' + lab);
          }
        });
      }
    }
    return separatedLabels.map((label, id) => (
      <Token
        key={id}
        className="y-fd-token y-fd-token--no-button y-fd-token--gap"
      >
        {label}
      </Token>
    ));
  };

  processStatus(status) {
    let type = 'warning';
    switch (status) {
      case 'INITIAL':
        return <Badge>{status}</Badge>;

      case 'READY':
        type = 'success';
        break;
      case 'UNKNOWN':
        type = 'warning';
        break;
      case 'FAILED':
        type = 'error';
        break;
      default:
        type = 'warning';
    }

    return <Badge type={type}>{status}</Badge>;
  }

  headerRenderer = applications => [
    'Name',
    'Description',
    'Labels',
    'APIs',
    'EventAPIs',
    'Status',
  ];
  rowRenderer = application => [
    <span
      className="link"
      onClick={() =>
        LuigiClient.linkManager()
          .fromClosestContext()
          .navigate(`/application/${application.id}`)
      }
    >
      {application.name}
    </span>,
    application.description,
    application.labels ? this.createLabels(application.labels) : '-',
    <Counter>{application.apis.totalCount}</Counter>,
    <Counter>{application.eventAPIs.totalCount}</Counter>,
    this.processStatus(application.status.condition),
  ];

  actions = [
    {
      name: 'Delete',
      handler: entry => {
        LuigiClient.uxManager()
          .showConfirmationModal({
            header: 'Remove application',
            body: `Are you sure you want to delete ${entry.name}?`,
            buttonConfirm: 'No',
            buttonDismiss: 'Also no',
          })
          .catch(() => {})
          .finally(() => {
            console.warn('As you wish, nothing will be removed');
          });
      },
    },
  ];

  render() {
    return (
      <Query query={GET_APPLICATIONS}>
        {({ loading, error, data }) => {
          if (loading) return 'Loading...';
          if (error) return `Error! ${error.message}`;
          const apps = data.applications.data;

          return (
            <>
              <GenericList
                extraHeaderContent={<CreateApplicationModal />}
                title="Applications"
                description="List of all aplications"
                actions={this.actions}
                entries={apps}
                headerRenderer={this.headerRenderer}
                rowRenderer={this.rowRenderer}
              />
            </>
          );
        }}
      </Query>
    );
  }
}
export default Applications;
