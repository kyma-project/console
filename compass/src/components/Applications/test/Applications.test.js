import React from 'react';
import Applications from '../Applications';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import { GET_APPLICATIONS_MOCK } from './mock';
import { shallow } from 'enzyme';

const wait = require('waait');

describe('Applications', () => {
  it('should render loading state initially', () => {
    const component = renderer.create(
      <MockedProvider mocks={GET_APPLICATIONS_MOCK} addTypename={false}>
        <Applications />
      </MockedProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('processStatus()', () => {
    const component = shallow(<Applications />);
    const processStatusFn = component.instance().processStatus;

    it('Renders proper <Badge> when no prop is provided', async () => {
      expect(processStatusFn()).toMatchSnapshot();
    });

    it('Renders proper <Badge> when INITIAL prop is provided', async () => {
      expect(processStatusFn('INITIAL')).toMatchSnapshot();
    });

    it('Renders proper <Badge> when ERROR prop is provided', async () => {
      expect(processStatusFn('ERROR')).toMatchSnapshot();
    });
  });
});
