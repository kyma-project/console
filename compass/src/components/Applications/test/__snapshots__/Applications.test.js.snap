// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications processStatus() Renders proper <Badge> when ERROR prop is provided 1`] = `
<Badge
  type="warning"
>
  ERROR
</Badge>
`;

exports[`Applications processStatus() Renders proper <Badge> when INITIAL prop is provided 1`] = `
<Badge>
  INITIAL
</Badge>
`;

exports[`Applications processStatus() Renders proper <Badge> when no prop is provided 1`] = `
<Badge
  type="warning"
/>
`;

exports[`Applications should render loading state initially 1`] = `"Loading..."`;
